src/devices_models/devices/common/timeseries_constraint.jl

```diff
@@ -1,31 +1,33 @@
 function lazy_lb(
     psi_container::PSIContainer,
-    ts_data::Vector{DeviceTimeSeries},
+    range_constraints::Vector{DeviceTimeSeriesRangeConstraint},
     cons_name::Symbol,
     var_name::Symbol,
 )
     time_steps = model_time_steps(psi_container)
-    names = (d.name for d in ts_data)
+    names = (get_name(x) for x in range_constraints)
     variable = get_variable(psi_container, var_name)
     lb_name = middle_rename(cons_name, PSI_NAME_DELIMITER, "lb")
     con_lb = add_cons_container!(psi_container, lb_name, names, time_steps)
 
-    for data in ts_data
-        if data.range.limits.min > -Inf && !isempty(data.range.additional_terms_lb)
-            con_lb[data.name, :].data .= JuMP.AffExpr(0.0)
+    for ts_range_constraint in range_constraints
+        rc_name = get_name(ts_range_constraint)
+        if ts_range_constraint.range_constraint.limits.min > -Inf &&
+           !isempty(ts_range_constraint.range_constraint.additional_terms_lb)
+            con_lb[rc_name, :].data .= JuMP.AffExpr(0.0)
```

Comment:
Note to me: Check that this is broadcasting properly, we used to have a problem with this in JuMP since it assigned every entry of the array the same object.